src/sx/src/StyleCollector.js

```diff
@@ -61,6 +61,9 @@ class StyleCollector {
           }
         } else if (maybeName.startsWith(':')) {
           // pseudo type (:hover, ::after, ...)
+          for (const key of Object.keys(maybeValue)) {
+            invariant(key.startsWith(':') === false, 'Nested pseudo classes are not allowed.');
```

Comment:
what about`::after:hover`?

Reply:
Do you mean when you write it as one selector? I actually didn't try it but I don't see a problem with that. It just cannot be nested. Will add some tests for it as well.

Reply:
Aha, of course, it is not nested in this case 🤦

Reply:
I added a test to cover this scenario:

```text
exports[`matches expected output: pseudo-classes-nested-2.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    ":hover::after": {
      "content": "this is OK"
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
._2BDpfz:hover::after {
  content: "this is OK";
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="_2BDpfz"

`;
```